18F/identity-idp
# LG-8886: Split percent logging for USPS proofing networking errors

## 🎫 Ticket

- [LG-8886](https://cm-jira.usa.gov/browse/LG-8886)
  - This PR only includes a portion of the changes for this ticket.

## 🛠 Summary of changes

- Log the percentage of networking errors for the USPS proofing results job separately from the percentage of other errors

## 📜 Testing Plan

- [x] Check that IPP is enabled and that the flag `in_person_enrollments_ready_job_enabled` is disabled
  - Note: This should work the same for `GetUspsWaitingProofingResultsJob` as for `GetUspsProofingResultsJob` if the flag is enabled. `GetUspsReadyProofingResultsJob` will only work for this test if the record already has the ready flag set.
- [x] Check that your local stack has `usps_mock_fallback` enabled
- [x] Create USPS enrollment (either with CLI tool or by completing in-person proofing flow)
  - ```sh
    bundle exec rails dev:random_in_person_users NUM_USERS='1' SCRYPT_COST='800$8$1$'
    ```
- [x] Update `UspsInPersonProofing::Mock::Proofer.request_proofing_results` to throw `Faraday::TimeoutError`
- [x] Run USPS proofing results job
  - ```sh
    rails c
    ```
  - ```ruby
     GetUspsProofingResultsJob.perform_now(Time.zone.now)
     ```
- Check that `log/events.log` includes the `GetUspsProofingResultsJob: Job completed` with the following fields:
  - [x] `enrollments_network_error` = `1`
  - [x] `percent_enrollments_network_error` = `100.00`
  - [x] `enrollments_errored` = `0`
  - [x] `percent_enrollments_errored` = `0.00`
- [x] Update `UspsInPersonProofing::Mock::Proofer.request_proofing_results` to throw `Faraday::ServerError`
- [x] Run USPS proofing results job
  - ```sh
    rails c
    ```
  - ```ruby
     GetUspsProofingResultsJob.perform_now(Time.zone.now)
     ```
- Check that `log/events.log` includes the `GetUspsProofingResultsJob: Job completed` with the following fields:
  - [x] `enrollments_network_error` = `0`
  - [x] `percent_enrollments_network_error` = `0.00`
  - [x] `enrollments_errored` = `1`
  - [x] `percent_enrollments_errored` = `100.00`
